library/camel-kamelets-utils/pom.xml

```diff
@@ -71,12 +71,22 @@
             <artifactId>camel-kafka</artifactId>
         </dependency>
 
-        <!-- AWS Dynamo DB camel component -->
+        <!-- Optional dependencies for data type conversion -->
         <dependency>
             <groupId>org.apache.camel</groupId>
             <artifactId>camel-aws2-ddb</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-aws2-s3</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-cloudevents</artifactId>
+            <!--<scope>provided</scope>-->
```

Comment:
Do we need to remove this?

Reply:
It is used in AWS S3 CloudEvents output data type.

The thing is `camel-cloudevents` is not available in camel-quarkus. Because of this the dependency is not able to resolve via CamelCatalog at runtime because the catalog only knows camel-quarkus extensions. When adding it here as a dependency it will be resolved as a transitive dependency using the Camel core version.

We should add it to camel-quarkus and use scope provided here so Camel K is able to resolve this dependency for bindings that actually use the data type.

Reply:
I had another look and the AWS S3 data type only uses some constants related to header names in camel-cloudevents. I will use plain String header names and remove the dependency for now.

Reply:
I removed the dependency in favor of using plain String constants

Reply:
there is a camel-cloudevents extension https://github.com/apache/camel-quarkus/tree/main/extensions/cloudevents